pkg/kmutex/kmutex.go

```diff
@@ -119,12 +139,15 @@ func (m *KMutex) Lock(k string) bool {
 }
 
 // Unlock release the lock.
+// If the key does not exist, return immediately.
+// Otherwise, fill up the chan to broadcast that someone released the lock.
 func (m *KMutex) Unlock(k string) {
 	m.Mutex.Lock()
 	defer m.Mutex.Unlock()
 
-	v, ok := m.keys[k]
+	v, ok := m.lockedKeys[k]
 	if ok && len(v.c) == 0 {
+		// filled up the chan to identify caller has released the lock.
 		v.c <- struct{}{}
 	}
 }
```

Comment:
Should it free key when not used anymore. Memory will increase when more and more keys are created. Some codes like:

```
if v.waits == 0 {
		// Free the lock if no one is holding or trying to hold.
		delete(m.lockedKeys, k)
	}
```